proxy/upgrade.go

```diff
@@ -0,0 +1,195 @@
+package proxy
+
+import (
+	"bufio"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// isUpgradeRequest returns true if and only if there is a "Connection"
+// key with the value "Upgrade" in Headers of the given request.
+func isUpgradeRequest(req *http.Request) bool {
```

Comment:
haha, didn't check what i am pasting in the rush :)
what i meant was this: http://www.iana.org/assignments/http-upgrade-tokens/http-upgrade-tokens.xhtml

whatever. with the flag, it'll be cool for now